datatypes/mcs_decimal.h

```diff
@@ -179,41 +179,89 @@ inline lldiv_t_128 lldiv128(const int128_t& dividend, const int128_t& divisor)
     return lldiv_t_128(dividend / divisor, dividend % divisor);
 }
 
+
+// TODO: derive it from TSInt64 eventually
+class TDecimal64
+{
+public:
+    int64_t value;
+public:
+    static constexpr uint8_t MAXLENGTH8BYTES = 23;
+
+public:
+    TDecimal64()
+     :value(0)
+    { }
+    explicit TDecimal64(int64_t val)
+     :value(val)
+    { }
+    // Divide to the scale divisor with rounding
+    int64_t toSInt64Round(uint32_t scale) const
+    {
+        double dscale = scale;
+        int64_t tmp = value / pow(10.0, dscale);
+        int lefto = (value - tmp * pow(10.0, dscale)) / pow(10.0, dscale - 1);
+        if (tmp >= 0 && lefto > 4)
+            return tmp + 1;
+        if (tmp < 0 && lefto < -4)
+            return tmp - 1;
+        return tmp;
+    }
+};
+
+
+class TDecimal128: public TSInt128
+{
+public:
+    static constexpr uint8_t MAXLENGTH16BYTES = TSInt128::maxLength();
+    static constexpr int128_t minInt128 = TFloat128::minInt128;
+    static constexpr int128_t maxInt128 = TFloat128::maxInt128;
+
+    static inline bool isWideDecimalNullValue(const int128_t& val)
+    {
+        return (val == TSInt128::NullValue);
+    }
+
+    static inline bool isWideDecimalEmptyValue(const int128_t& val)
+    {
+        return (val == TSInt128::EmptyValue);
+    }
+
+    static inline void setWideDecimalNullValue(int128_t& val)
+    {
+        val = TSInt128::NullValue;
+    }
+
+    static inline void setWideDecimalEmptyValue(int128_t& val)
+    {
+        val = TSInt128::EmptyValue;
+    }
+
+public:
+    TDecimal128()
+    { }
+    explicit TDecimal128(const int128_t &val)
+       :TSInt128(val)
+    { }
+    explicit TDecimal128(const TSInt128& val)
+       :TSInt128(val)
+    { }
+    explicit TDecimal128(const int128_t* valPtr)
+       :TSInt128(valPtr)
+    { }
+};
+
+
 // @brief The class for Decimal related operations
 // The class contains Decimal related operations are scale and
 // precision aware.
 // This class will inherit from:
 //      DecimalMeta class that stores scale and precision
 //      Storage classes TSInt128 and int64
```

Comment:
Since we don't directly inherit from the storage classes TSInt128 and TSInt64 (future), can we fix this comment? I think we can simple remove this above line, and just keep the "DecimalMeta" comment and change it to "In future, this class will inherit from DecimalMeta that stores scale and precision".